### PR TITLE
lesson 4 - mail_room_part2

### PR DESCRIPTION
**Pull request by april301975**

### CODE REVIEW

File: students/Thai_H/lesson4/mail_room_part2.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+def init_donor_list():
+    # create initial list of donors and their donations
+    global list_of_donors
+    list_of_donors = [
+                {'name': 'William Boeing', 'donation': [12.86, 2.00, 3.00, 4.00, 5.00]},
+                {'name': 'Steve Jobs', 'donation': [6.00, 7.00, 8.00]},
+                {'name': 'Paul Allen', 'donation': [9.00, 10.00, 11.00]},
+                {'name': 'Charles Flint', 'donation': [12.00, 13.00, 14.00]},
+                {'name': 'Thomas Edison', 'donation': [15.00, 16.00, 17.00]}
+            ]
```
Comment (charlesrobison) [inline students/Thai_H/lesson4/mail_room_part2.py]: This is a list of dictionaries - for this portion of the assignment, you'll want one dictionary with names as the keys and donations as the values.

File: students/Thai_H/lesson4/mail_room_part2.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        for each_donor in list_of_donors:
+            if donor_name == each_donor['name']:  #[0] is existing donor name
+                each_donor['donation'].append(float(donation_amt))
+
+    # New donor
+    else:
+        new_donor_name_amt = {'name':donor_name,'donation':[float(donation_amt)]}
+        list_of_donors.append(new_donor_name_amt)
+    # send thank you letter to the the donor name user just entered
+    send_thank_you_letter(donor_name, donation_amt)
```
Comment (charlesrobison) [inline students/Thai_H/lesson4/mail_room_part2.py]: This doesn't appear to be functioning correctly - isn't adding new donors or new donations to existing amounts.

File: students/Thai_H/lesson4/mail_room_part2.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+#----------------------------------------------------------------------------
+def thank_you_menu_selection():
+    user_selection = input('\n\nTHANK-YOU MENU\nChoose from one of options below:\n\
+        L) List current donors\n\
+        T) Send Thank-you letter (or Add donor and Send one)\n\
+        F) Create Thank-you files\n\
+        M) Back to MAIN MENU\n\
+        Q) Quit the program\n\
+        Please type L, N, or B: ')
```
Comment (charlesrobison) [inline students/Thai_H/lesson4/mail_room_part2.py]: Why only L, N, or B?  Additional options above.

**Comment by natasha-aleksandrova** [comment]
@april301975 merging but needs additional work.

**Comment by natasha-aleksandrova** [comment]
@april301975 merging but needs additional work.

**Pull request merged**